### ISSUE TITLE
moto lambda response Payload is the log content, not function response data

### ISSUE THREAD

**Issue by dazza-codes** [spulec/moto#2886]
See all details in https://github.com/aio-libs/aiobotocore/issues/793
- https://github.com/aio-libs/aiobotocore/pull/796 demonstrates the problem

Although I don't know how to reconcile that with the moto unit tests, i.e.
- https://github.com/spulec/moto/blob/master/tests/test_awslambda/test_lambda.py#L95-L125
- this appears to distinguish between the logs and payload

**Comment by bblommers** [comment]
Hi @dazza-codes, it looks like that issue was fixed in this PR: https://github.com/spulec/moto/pull/2669

Can you repro this issue with the latest version? (https://pypi.org/project/moto/#history)

**Comment by thehesiod** [comment]
Oh you did already, sorry didn't notice from my phone :)

**Comment by dazza-codes** [comment]
Code and tests to replicate this are in https://gitlab.com/dazza-codes/aio-aws/-/merge_requests/8

### released versions in venv
```
$ pip freeze | grep -i oto
aiobotocore==0.12.0
boto==2.49.0
boto3==1.12.3
botocore==1.15.3
moto==1.3.14
```

### test failure debugging

```
tests/test_aio_aws_lambda.py:154: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
aio_aws/aio_aws_lambda.py:163: in invoke
    await self.read_response()  # updates self.content
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = AWSLambdaFunction(name='lambda_dev', type='RequestResponse', log_type='None', context=None, payload=b'{"i": 1}', quali...s\tBilled Duration: 100 ms\tMemory Size: 128 MB\tMax Memory Used: 26 MB\t\x1b[0m\n\n{"statusCode":200,"body":{"i":1}}')

    async def read_response(self):
        """
        Asynchronous coroutine to read a lambda response; this
        updates the ``content`` or the ``error`` values.
    
        :raises: botocore.exceptions.ClientError, botocore.exceptions.ParamValidationError
        """
    
        if self.response and response_success(self.response):
    
            log_result = self.response.get("LogResult")
            if log_result:
                self.logs = base64.b64decode(log_result)
    
            response_payload = self.response.get("Payload")
            if response_payload:
    
                async with response_payload as stream:
                    data = await stream.read()
    
                body = data.decode()
    
>               assert False  # for debugging
E               AssertionError

aio_aws/aio_aws_lambda.py:207: AssertionError
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> entering PDB >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB post_mortem >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /home/dlweber/src/aio-aws/aio_aws/aio_aws_lambda.py(207)read_response()
-> assert False
(Pdb) self.logs
b'\x1b[32mSTART RequestId: e1c62ace-c0ff-194f-8896-1a80ed362f7b Version: 1\x1b[0m\nevent: {\'i\': 1}\n\x1b[32mEND RequestId: e1c62ace-c0ff-194f-8896-1a80ed362f7b\x1b[0m\n\x1b[32mREPORT RequestId: e1c62ace-c0ff-194f-8896-1a80ed362f7b\tInit Duration: 532.85 ms\tDuration: 3.41 ms\tBilled Duration: 100 ms\tMemory Size: 128 MB\tMax Memory Used: 26 MB\t\x1b[0m\n\n{"statusCode":200,"body":{"i":1}}'
(Pdb) body
'\x1b[32mSTART RequestId: e1c62ace-c0ff-194f-8896-1a80ed362f7b Version: 1\x1b[0m\nevent: {\'i\': 1}\n\x1b[32mEND RequestId: e1c62ace-c0ff-194f-8896-1a80ed362f7b\x1b[0m\n\x1b[32mREPORT RequestId: e1c62ace-c0ff-194f-8896-1a80ed362f7b\tInit Duration: 532.85 ms\tDuration: 3.41 ms\tBilled Duration: 100 ms\tMemory Size: 128 MB\tMax Memory Used: 26 MB\t\x1b[0m\n\n{"statusCode":200,"body":{"i":1}}'
(Pdb) 
```



When it runs against an actual AWS lambda function (i.e. live test), it has a very different response that works.

```
Test async lambda functions
[INFO]  2020-04-15T01:11:05.215Z  aio-aws:invoke:165  AWS Lambda (lambda_dev) invoked OK

{'FunctionName': 'lambda_dev', 'InvocationType': 'RequestResponse', 'LogType': 'None', 'Payload': b'{"i": 0}'}

{'ResponseMetadata': {'RequestId': '8fd8d972-f00c-4d5f-95b7-723d896fe133', 'HTTPStatusCode': 200, 'HTTPHeaders': {'date': 'Wed, 15 Apr 2020 01:11:05 GMT', 'content-type': 'application/json', 'content-length': '63', 'connection': 'keep-alive', 'x-amzn-requestid': '8fd8d972-f00c-4d5f-95b7-723d896fe133', 'x-amzn-remapped-content-length': '0', 'x-amz-executed-version': '$LATEST', 'x-amzn-trace-id': 'root=1-5e965f29-b530e6c8868242e0385652d0;sampled=0'}, 'RetryAttempts': 0}, 'StatusCode': 200, 'ExecutedVersion': '$LATEST', 'Payload': <StreamingBody at 0x7f24ee2dd948 for ClientResponseContentProxy at 0x7f24ee2dd908>}

{'statusCode': 200, 'body': '"0, 1, 2, 3, 4, 5, 6, 7, 8, 9"'}

None
1 lambdas finished in 0.30 seconds.
```

Things to note:
- `StatusCode` is 200 from AWS, 202 from moto
- `content-type` is json from AWS, html from moto
- with `LogType == None`, AWS has no logs, moto has logs (same data in the body/content too)

**Comment by bblommers** [comment]
This has now been fixed in moto >= 1.3.15.dev859
